lib/Service/ApiService.php

```diff
@@ -113,12 +113,12 @@ public function create($fileId = null, $filePath = null, $token = null, $guestNa
 			$readOnly = $this->documentService->isReadOnly($file, $token);
 
 			$this->sessionService->removeInactiveSessions($file->getId());
-			$activeSessions = $this->sessionService->getActiveSessions($file->getId());
+			$remainingSessions = $this->sessionService->getAllSessions($file->getId());
 			$freshSession = false;
-			if ($forceRecreate || count($activeSessions) === 0) {
+			if ($forceRecreate || count($remainingSessions) === 0) {
```

Comment:
Not sure I understand the impact of this change: we only create a fresh session now if "no session was found", instead of "no *active* session was found" before, right? So we will always try to restore sessions, even if they're not active anymore?

Reply:
Yes. Sessions become inactive after 5 minutes. But there may still be steps that were created during the session but never persisted. All the cleanup functions will leave those steps and their session in place (unless `forceReload` is set). This change makes it so we still recover these changes after more than 5 minutes.